subprojects/scala/src/main/java/org/gradle/api/internal/tasks/scala/ZincScalaCompilerFactory.java

```diff
@@ -113,43 +130,72 @@ static ZincScalaCompiler getCompiler(CacheRepository cacheRepository, HashedClas
         return new ZincScalaCompiler(scalaInstance, scalaCompiler, new AnalysisStoreProvider());
     }
 
-    private static ClassLoader getClassLoader(ClassPath classpath) {
+    private static ClassLoader getClassLoader(ClassPath classpath, ClassLoader parent) {
         try {
             List<URL> urls = new ArrayList<URL>();
             for (File file : classpath.getAsFiles()) {
-                urls.add(file.toURI().toURL());
+                // Having the bridge in the classloader breaks zinc
+                if (!file.toString().contains("scala3-sbt-bridge")) {
+                    urls.add(file.toURI().toURL());
+                }
             }
-            return new URLClassLoader(urls.toArray(new URL[0]));
+            return new URLClassLoader(urls.toArray(new URL[0]), parent);
```

Comment:
Looking at the class hierarchy, passing `null` here is not the same as calling the single argument constructor - in the later case you implicitly get the system classloader as the parent.

Reply:
Och! I was sure it is the same, I need to finally download JDK 16 and test it out.

Reply:
And indeed fixing that call fixes some of the test failures.